tests/PathTest.php

```diff
@@ -945,4 +945,106 @@ public function testIsBasePath($path, $ofPath, $result)
     {
         $this->assertSame($result, Path::isBasePath($path, $ofPath));
     }
+
+    public function provideCombineTests()
+    {
+        $object = new ObjectConvertibleToString('object/path');
+
+        return array(
+            array(null, null, null),
+            array('/path/to/test', null, '/path/to/test'),
+            array('/path/to//test', null, '/path/to/test'),
+            array(null, '/path/to/test', '/path/to/test'),
+            array(null, '/path/to//test', '/path/to/test'),
+
+            array('/path/to/test', 'subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', 'subdir', '/path/to/test/subdir'),
+            array('/path/to/test', '/subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', '/subdir', '/path/to/test/subdir'),
+            array('/path/to/test', './subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', './subdir', '/path/to/test/subdir'),
+            array('/path/to/test/', '../parentdir', '/path/to/parentdir'),
+            array('/path/to/test', '../parentdir', '/path/to/parentdir'),
+            array('path/to/test/', '/subdir', 'path/to/test/subdir'),
+            array('path/to/test', '/subdir', 'path/to/test/subdir'),
+            array('../path/to/test', '/subdir', '../path/to/test/subdir'),
+            array('path', '../../subdir', '../subdir'),
+            array('/path', '../../subdir', '/subdir'),
+            array('../path', '../../subdir', '../../subdir'),
+
+            array('/path/to/test', 123, '/path/to/test/123'),
+            array('/path/to/test', $object, '/path/to/test/object/path'),
+
+            array(array('/path/to/test', 'subdir'), null, '/path/to/test/subdir'),
+            array(array('/path/to/test', '/subdir'), null, '/path/to/test/subdir'),
+            array(array('/path/to/test/', 'subdir'), null, '/path/to/test/subdir'),
+            array(array('/path/to/test/', '/subdir'), null, '/path/to/test/subdir'),
```

Comment:
What happens when you pass an array with just one/no element? I think you should add test cases for that.